src/Main.gren

```diff
@@ -103,8 +136,78 @@ subscriptions : Model -> Sub Msg
 subscriptions model =
     when model.server is
         Just server ->
-            Http.onRequest server <| \req res -> 
+            HttpServer.onRequest server <| \req res -> 
                 GotRequest { request = req, response = res }
 
         Nothing ->
             Sub.none
+
+
+-- TODO: move to module
+
+migrations : Array { statement : String, parameters : Array Db.Encode.Value }
+migrations =
+    [ { statement = "PRAGMA foreign_keys = ON"
+      , parameters = []
+      }
+
+    , { statement = "PRAGMA busy_timeout = 5000"
+      , parameters = []
+      }
+
+    , { statement = 
+        """
+        CREATE TABLE IF NOT EXISTS user (
+            id INTEGER PRIMARY KEY,
+            email TEXT NOT NULL UNIQUE,
+            role TEXT
+        ) STRICT
+        """
+      , parameters = []
+      }
+
+    , { statement =
+        """
+        CREATE TABLE IF NOT EXISTS session (
+            id INTEGER PRIMARY KEY,
+            created INTEGER NOT NULL,
+            user_id INTEGER NOT NULL REFERENCES user(id),
+
+            -- Token used in email validation link.
+            validation_token TEXT NOT NULL UNIQUE,
+
+            -- Create and show this when user follows email validation link.
+            -- User will provide this to the CLI.
+            validation_code TEXT UNIQUE,
+
+            -- Token used by the CLI to fetch session token.
+            fetch_session_token TEXT NOT NULL UNIQUE,
+
+            -- Create and return this when CLI posts with validation_code and fetch_session_token
+            session_token TEXT UNIQUE
+        ) STRICT
+        """
+      , parameters = []
+      }
+
+    , { statement =
+        """
+        CREATE TABLE IF NOT EXISTS publishing_identity (
+            id INTEGER PRIMARY KEY,
+            name TEXT NOT NULL UNIQUE
+        ) STRICT
+        """
+      , parameters = []
+      }
+
+    , { statement =
+        """
+        CREATE TABLE IF NOT EXISTS publishing_identity_users (
+            user_id INTEGER NOT NULL REFERENCES user(id),
+            publishing_identity_id INTEGER NOT NULL REFERENCES publishing_identity(id),
+            role_id INTEGER NOT NULL REFERENCES role(id)
```

Comment:
role table doesn't exist anymore, right?

I guess this is where `role` should be (it's currently on the user table).